### PR TITLE
Complete support for Docker and Github Codespace 

### PR DESCRIPTION
**Pull request by zhangwei217245**
Now we can run our dev environment with the support of Docker and we can save some node hours for NERSC.

### CODE REVIEW

**Comment by zhangwei217245** [comment]
Tested for NERSC-ci: 

image
image
image

**Review by houjun — APPROVED** [review]

File: .devcontainer/devcontainer.Dockerfile [hunk 1]
```diff
@@ -1,95 +1,8 @@
...
-    make -j && make install && \
-    ctest
-
-# Set the environment variables
-ENV LD_LIBRARY_PATH="$MERCURY_DIR/lib:$LD_LIBRARY_PATH"
-ENV PATH="$MERCURY_DIR/include:$MERCURY_DIR/lib:$PATH"
-RUN echo 'export LD_LIBRARY_PATH=$MERCURY_DIR/lib:$LD_LIBRARY_PATH' >> $WORK_SPACE/pdc_env.sh \
-    echo 'export PATH=$MERCURY_DIR/include:$MERCURY_DIR/lib:$PATH' >> $WORK_SPACE/pdc_env.sh
+# FROM ubuntu:jammy
+FROM zhangwei217245/pdc_dev_base:latest
```
Comment (jeanbez) [inline .devcontainer/devcontainer.Dockerfile]: Let's move this to our own registry

File: .docker/run_dev_base.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+DEFAULT_WORKSPACE=/workspaces/pdc
+
+LAST_PDC_DEV_CONTAINER=$(docker ps -a | grep pdc_dev_base | head -1 | awk '{print $NF}') 
+
+# if this is empty, then we need to create a new container
+if [ -z "$LAST_PDC_DEV_CONTAINER" ]; then
+    echo "No existing pdc_dev_base container found, creating a new one"
+    docker image rm -f zhangwei217245/pdc_dev_base:latest
```
Comment (jeanbez) [inline .docker/run_dev_base.sh]: Move to new URL

File: .docker/run_dev_base.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    docker create -it -v $(pwd):${DEFAULT_WORKSPACE} -w ${DEFAULT_WORKSPACE} --entrypoint /bin/bash zhangwei217245/pdc_dev_base:latest
```
Comment (jeanbez) [inline .docker/run_dev_base.sh]: Here are well

**Review by jeanbez — CHANGES_REQUESTED** [review]

File: .docker/run_dev_base.sh (hunk 3 above)
Comment (zhangwei217245) [reply .docker/run_dev_base.sh]: will make changes!

File: .docker/run_dev_base.sh (hunk 2 above)
Comment (zhangwei217245) [reply .docker/run_dev_base.sh]: addressed

Comment (zhangwei217245) [reply .docker/run_dev_base.sh]: addressed

File: .docker/run_dev_base.sh (hunk 3 above)
Comment (zhangwei217245) [reply .docker/run_dev_base.sh]: addressed

**Review by jeanbez — APPROVED** [review]

**Pull request merged**